sdk/src/main/scala/com/horizen/api/http/SidechainTransactionApiRoute.scala

```diff
@@ -153,7 +153,7 @@ case class SidechainTransactionApiRoute(override val settings: RESTApiSettings,
             val outputs: JList[BoxData[Proposition, Box[Proposition]]] = new JArrayList()
             body.regularOutputs.foreach(element =>
               outputs.add(new ZenBoxData(
-                PublicKey25519PropositionSerializer.getSerializer.parseBytes(BytesUtils.fromHexString(element.publicKey)),
+                PublicKey25519PropositionSerializer.getSerializer.parseBytesAndCheck(BytesUtils.fromHexString(element.publicKey)),
```

Comment:
I assume it may exists more places like this for `Secp256k1`, `VRF` and `Schnorr` keys.
Let's add similar verification logic there.

Reply:
For Secp256k1 public key this logic can not be implemented cause we use hash of public key for the proposition.

For VRF and Schnorr public key this type of validation are implemented.